python/ray/runtime_env/runtime_env.py

```diff
@@ -42,21 +42,31 @@ class RuntimeEnvConfig(dict):
         eager_install: Indicates whether to install the runtime environment
             on the cluster at `ray.init()` time, before the workers are leased.
             This flag is set to `True` by default.
+        disable_cache: Indicates whether to cache the runtime environment on the
```

Comment:
I also think this flag is too global that affects all the runtime env setups and disables cache for successful setups as well. What we want is disabling cache for transient errors. We can define a few http code as transient errors and don't cache the result if we receive those http codes.

Reply:
What's the issue with that?

Reply:
The URI is same but the sha256 not, with flag can resolved this ASAP.

Reply:
> The URI is same but the sha256 not

That's a separate issue. Currently we expect a different URI for a different content.

Even if we want `disable_cache` option, it shouldn't be global but per runtime env.

Reply:
Yeah. The `disable_cache` is not global, we read it from per request.

Reply:
Yes. It coverage the case we discussed above. Maybe we need two PRs?

- [ ]  Catch the status code to handle the failed case
- [ ]  Refactor the cache key use sha256 not URI

Reply:
> Refactor the cache key use sha256 not URI

We already have a separate GH issue for this change. It would be great if you can also work on it :)

Reply:
> > Refactor the cache key use sha256 not URI
> 
> We already have a separate GH issue for this change. It would be great if you can also work on it :)

Assign to me, thx!

Reply:
I tagged you on few related issues.